controllers/helmrepository_controller.go

```diff
@@ -209,11 +211,17 @@ func (r *HelmRepositoryReconciler) reconcile(ctx context.Context, repository sou
 		return sourcev1.HelmRepositoryNotReady(repository, sourcev1.IndexationFailedReason, err.Error()), err
 	}
 
-	// return early on unchanged generation
+	// return early on unchanged index
+	indexContent, err := yaml.Marshal(chartRepo.Index)
+	if err != nil {
+		return sourcev1.HelmRepositoryNotReady(repository, sourcev1.StorageOperationFailedReason, err.Error()), err
+	}
+	sum := sha256.Sum256(indexContent)
+	hash := hex.EncodeToString(sum[:])
```

Comment:
It would probably be better to make use of the `Storage#Checksum` method here.

Also, to avoid loading things in memory twice, the code on L291 should reuse the bytes from `indexContent` instead of marshaling again.

Reply:
Thanks for the review! I have to admit my Golang skills are quite limited. :)